src/utils.ts

```diff
@@ -0,0 +1,111 @@
+import { Address } from "@multiversx/sdk-core/out/address";
+import Client from "@walletconnect/sign-client";
+import { EngineTypes, SessionTypes } from "@walletconnect/types";
+
+import {
+  WALLETCONNECT_MULTIVERSX_METHODS,
+  WALLETCONNECT_MULTIVERSX_NAMESPACE,
+} from "./constants";
+
+import { WalletConnectV2ProviderErrorMessagesEnum } from "./errors";
+import { Logger } from "./logger";
+
+export interface ConnectParamsTypes {
+  topic?: string;
+  events?: SessionTypes.Namespace["events"];
+  methods?: string[];
+}
+
+export function getCurrentSession(
+  client: Client,
+  chainId: string
+): SessionTypes.Struct {
+  if (typeof client === "undefined") {
+    throw new Error(WalletConnectV2ProviderErrorMessagesEnum.notInitialized);
+  }
+
+  const acknowledgedSessions = client
+    .find(getConnectionParams(chainId))
+    .filter((s) => s.acknowledged);
+
+  if (acknowledgedSessions.length > 0) {
+    const lastKeyIndex = acknowledgedSessions.length - 1;
+    const session = acknowledgedSessions[lastKeyIndex];
+
+    return session;
+  } else if (client.session.length > 0) {
```

Comment:
`else` statements can be removed (can also stay as they are).

Reply:
Updated